src/features/profile/index.tsx

```diff
@@ -0,0 +1,223 @@
+"use client";
+
+import { Avatar, AvatarImage } from "@/components/ui/avatar";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
+import { client } from "@/lib/hono";
+import type { InferResponseType } from "hono";
+import { useRef, useState } from "react";
+import { toast } from "sonner";
+
+const gradeOptions = ["B1", "B2", "B3", "B4", "M1", "M2", "D1", "D2", "D3"] as const;
+
+type Grade = (typeof gradeOptions)[number];
+
+const groupOptions = ["行政", "金融", "LLM", "スポーツ", "マルチモーダル", "アプリ開発", "その他"] as const;
+
+type Group = (typeof groupOptions)[number];
+
+const req = client.api.users[":user_id"].$get;
+
+type Props = {
+  userId: string;
+  data: InferResponseType<typeof req, 200>;
+};
+
+export default function ProfilePage({ userId, data }: Props) {
+  const [name, setName] = useState(data.name ?? "");
+  const [displayname, setDisplayName] = useState(data.displayName ?? "");
+  const [grade, setGrade] = useState<Grade | "">(
+    gradeOptions.includes(data.grade as Grade) ? (data.grade as Grade) : "",
+  );
+  const [group, setGroup] = useState<Group | "">(
+    groupOptions.includes(data.group as Group) ? (data.group as Group) : "",
+  );
+  const [imageUrl, setImageUrl] = useState(data.image ?? "/logo.png");
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const savedValues = useRef({
+    name,
+    displayname,
+    grade,
+    group,
+    imageUrl,
+  });
+
+  const handleAvatarClick = () => {
+    fileInputRef.current?.click();
+  };
+
+  const handleFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    const toastId = toast.loading("アップロード中...");
+    const file = event.target.files?.[0];
+    if (!file) return;
+
+    const filename = `${userId}_${Date.now()}`;
```

Comment:
userIdごとのフォルダに分けておくと探しやすくなるので、`/`で区切るようにお願いします
```suggestion
    const filename = `${userId}/${Date.now()}`;
```